clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -63,11 +63,9 @@ public Principal principal() throws IOException {
      * Does handshake of transportLayer and Authentication using configured authenticator
      */
     public void prepare() throws IOException {
-        if (transportLayer.ready() && authenticator.complete())
-            return;
         if (!transportLayer.ready())
             transportLayer.handshake();
-        if (transportLayer.ready() && !authenticator.complete())
+        else if (transportLayer.ready() && !authenticator.complete())
```

Comment:
I agree that it's not needed in its current state, but it makes sense with the code as it was before, that is:

```
        if (!transportLayer.ready())
            transportLayer.handshake();
        if (transportLayer.ready() && !authenticator.complete())
            authenticator.authenticate();
```

Which version do we prefer?